### PR TITLE
Enable tests to validate dist build contents

### PR DESCRIPTION
**Pull request by lioman**
# Pull Request Checklist

These test should ensure that all contents are there and one can not misconfigure it in the future.

- [x] Ensured **tests pass** and (if applicable) updated functional test output
- [x] Conformed to **code style guidelines** by running appropriate linting tools
- [x] Added **tests** for changed code
- [ ] Updated **documentation** for changed code

### CODE REVIEW

File: pelican/tests/build_test/test_build_files.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        files_list = tarfile.open(dist, "r:gz").getnames()
+        dir_matcher = ""
+        if expected_file.endswith("/"):
+            dir_matcher = ".*"
+        filtered_values = list(
+            filter(
+                lambda v: match(f"^pelican-\d.\d.\d/{expected_file}{dir_matcher}$", v),
+                files_list,
+            )
+        )
```
Comment (avaris) [inline pelican/tests/build_test/test_build_files.py:66]: I don't like `filter` in terms of readability:

```python
pattern = f"^pelican-\d.\d.\d/{expected_file}{dir_matcher}$"
filtered_values = [file for file in files_list if match(pattern, file)]
```

Comment (lioman) [reply pelican/tests/build_test/test_build_files.py]: Even if I like `filter` in that case a list comprehension is really more readable

File: pelican/tests/build_test/test_build_files.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        ("THANKS"),
+        ("README.rst"),
+        ("CONTRIBUTING.rst"),
+        ("docs/changelog.rst"),
+        ("samples/"),
+    ],
+)
+def test_sdist_contents(pytestconfig, expected_file):
+    """
+    This test, should test the contents of the wheel to make sure,
```
Comment (avaris) [inline pelican/tests/build_test/test_build_files.py:51]: `wheel` -> `sdist`

And, I'd remove `,`s. Same for the test above.

**Review by avaris — COMMENTED** [review]
Just noticed some minor docstring corrections, besides that this looks good to me.

**Review by avaris — APPROVED** [review]
Thanks!

**Pull request merged**